evm_loader/program/src/executor/state.rs

```diff
@@ -467,3 +473,21 @@ impl<'a, B: AccountStorage> Database for ExecutorState<'a, B> {
         self.backend.contract_chain_id(contract).await
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use hex::FromHex;
+
+    // https://eips.ethereum.org/EIPS/eip-1052
+    #[test]
+    fn test_keccak_hash_empty_slice() {
```

Comment:
This should be moved to `opcode` module.

Reply:
This test tests whether the hash is implemented correctly. I don't think we should be verifying that given that it is provided by Solana. What do you think?

Reply:
I do think you could make a case to test if our implementation is correct, I.E. if we return the correct hash for:

1. a non-existing account
2. an existing balance account (without code)
3. an existing program account (with non-zero length code)
4. an existing program account(with zero length code) <- if this is possible

Reply:
Yes, I added the test more for documentation purposes, since the hash of an empty slice being something apparently random is unexpected. At least it was for me. You can remove it if you want.

Yes, I agree some unit tests for these cases should be added.

Reply:
Any suggestions on how I might provide a `Database` implementation with a specific configuration of accounts? I'll need that in my tests. Could create a mock implementation by hand, through a lib, or use a real implementation with mock data.

Reply:
I think I did it already in #236. You should help with reviewing that PR so I can merge it faster.

Reply:
I've added tests!